airflow/providers/amazon/aws/triggers/neptune.py

```diff
@@ -0,0 +1,115 @@
+    :param waiter_max_attempts: The maximum number of attempts to be made.
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    :param region_name: AWS region name (example: us-east-1)
+    """
+
+    def __init__(
+        self,
+        *,
+        db_cluster_id: str,
+        waiter_delay: int = 30,
+        waiter_max_attempts: int = 60,
+        aws_conn_id: str | None = None,
+        region_name: str | None = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            serialized_fields={"db_cluster_id": db_cluster_id},
+            waiter_name="cluster_available",
+            waiter_args={"DBClusterIdentifier": db_cluster_id},
+            failure_message="Neptune cluster failed",
+            status_message="Status of Neptune cluster is",
+            status_queries=["DBClusters[0].Status"],
+            return_key="db_cluster_id",
+            return_value=db_cluster_id,
+            waiter_delay=waiter_delay,
+            waiter_max_attempts=waiter_max_attempts,
+            aws_conn_id=aws_conn_id,
+            **kwargs,
+        )
+
+    def hook(self) -> AwsGenericHook:
+        return NeptuneHook(
+            aws_conn_id=self.aws_conn_id,
+            region_name=self.region_name,
+            verify=self.verify,
+            config=self.botocore_config,
+        )
+
+
+class NeptuneClusterStoppedTrigger(AwsBaseWaiterTrigger):
+    """
+    Triggers when a Neptune Cluster is stopped.
+
+    :param db_cluster_id: Cluster ID to poll.
+    :param waiter_delay: The amount of time in seconds to wait between attempts.
+    :param waiter_max_attempts: The maximum number of attempts to be made.
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    :param region_name: AWS region name (example: us-east-1)
+    """
+
+    def __init__(
+        self,
+        *,
+        db_cluster_id: str,
+        waiter_delay: int = 30,
+        waiter_max_attempts: int = 60,
+        aws_conn_id: str | None = None,
+        region_name: str | None = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            serialized_fields={"db_cluster_id": db_cluster_id},
+            waiter_name="cluster_stopped",
+            waiter_args={"DBClusterIdentifier": db_cluster_id},
+            failure_message="Neptune cluster failed",
+            status_message="Status of Neptune cluster is",
+            status_queries=["DBClusters[0].Status"],
+            return_key="db_cluster_id",
+            return_value=db_cluster_id,
+            waiter_delay=waiter_delay,
+            waiter_max_attempts=waiter_max_attempts,
+            aws_conn_id=aws_conn_id,
+            **kwargs,
+        )
+
+    def hook(self) -> AwsGenericHook:
+        return NeptuneHook(
+            aws_conn_id=self.aws_conn_id,
+            region_name=self.region_name,
+            verify=self.verify,
+            config=self.botocore_config,
+        )
```

Comment:
I am not sure whether creating yet another `NeptuneClusterGenricTrigger` makes sense. But by doing so, we could remove one duplicate `hook` method

Reply:
I'm not sure I follow, what do you mean?

Reply:
```
class NeptuneClusterGenricTrigger:
```suggestion
class NeptuneClusterGenricTrigger
    def hook(self) -> AwsGenericHook:
        return NeptuneHook(
            aws_conn_id=self.aws_conn_id,
            region_name=self.region_name,
            verify=self.verify,
            config=self.botocore_config,
        )
```
```

Reply:
Gotcha. Not sure if it would make sense or not. `AwsBaseWaiterTrigger` requires subclasses to implement `hook`, not sure if adding the additional inheritance gains much.